src/vs/workbench/api/node/extHost.api.impl.ts

```diff
@@ -255,6 +255,18 @@ export function createApiFactory(initData: IInitData, threadService: IThreadServ
 			}
 		};
 
+		const emptyMessageOptions: vscode.MessageOptions = Object.create(null);
+
+		function parseMessageArguments(args: any[]): { options: vscode.MessageOptions; items: any[]; } {
+			const [first, ...rest] = args;
+
+			if (typeof first === 'string' || (first && first.title)) {
+				return { options: emptyMessageOptions, items: args };
+			} else {
+				return { options: first || emptyMessageOptions, items: rest };
+			}
+		}
```

Comment:
I like the code but I don't like that it is placed here. Instead I'd propose to move it into the `extHostMessageService` and make its `showMessage`-function like: `showMessage(sev, message, optionOrFirstItem, ...restItem)`. That would encapsulate things better IMO